### PR TITLE
V7 result

### PR DESCRIPTION
**Pull request by mxcl**

### CODE REVIEW

**Comment by mxcl** [comment]
Merge after v7 branch.

File: Tests/Core/ResolverTests.swift [hunk 1]
```diff
@@ -131,8 +131,8 @@ class WrapTests: XCTestCase {
     }
 
     func testIsFulfilled() {
```
Comment (mmatoszko) [inline Tests/Core/ResolverTests.swift]: Shouldn't the test name also change to `testIsSuccess()`?

Comment (mxcl) [reply Tests/Core/ResolverTests.swift]: You’re right, thanks!

File: Sources/Promise.swift [hunk 2]
```diff
@@ -39,12 +39,12 @@ public final class Promise<T>: Thenable, CatchMixin {
           }
      */
     public class func value(_ value: T) -> Promise<T> {
-        return Promise(box: SealedBox(value: .fulfilled(value)))
+        return Promise(box: SealedBox(value: .success(value)))
     }
 
     /// Initialize a new rejected promise.
```
[the remark this answers is not in the document]
Comment (mxcl) [reply Sources/Promise.swift]: Fulfilled & rejected are established Promise nomenclature. The fact that we use a Result type underneath the promise is an implementation detail. So I'm not sure we need to change all these examples.

[the remark this answers is not in the document]
Comment (mmatoszko) [reply Sources/Promise.swift]: Got it, thanks :)

**Pull request merged**